src/config.rs

```diff
@@ -333,32 +524,29 @@ impl Config {
     ) -> Result<T::Value> {
         config_map!(self, opts => ::internal::deserialize_from_custom_seed(seed, reader, opts))
     }
+}
 
-    /// Executes the acceptor with a serde::Deserializer instance.
-    /// NOT A PART OF THE STABLE PUBLIC API
-    #[doc(hidden)]
-    pub fn with_deserializer<'a, A, R>(&self, reader: R, acceptor: A) -> A::Output
-    where
-        A: DeserializerAcceptor<'a>,
-        R: BincodeRead<'a>,
-    {
-        config_map!(self, opts => {
-            let mut deserializer = ::de::Deserializer::new(reader, opts);
-            acceptor.accept(&mut deserializer)
-        })
+mod internal {
+    use super::*;
+    use byteorder::ByteOrder;
+
+    pub trait Options: Clone {
+        type Limit: SizeLimit + 'static;
+        type Endian: BincodeByteOrder + 'static;
+
+        fn limit(&mut self) -> &mut Self::Limit;
     }
 
-    /// Executes the acceptor with a serde::Serializer instance.
-    /// NOT A PART OF THE STABLE PUBLIC API
-    #[doc(hidden)]
-    pub fn with_serializer<A, W>(&self, writer: W, acceptor: A) -> A::Output
-    where
-        A: SerializerAcceptor,
-        W: Write,
-    {
-        config_map!(self, opts => {
-            let mut serializer = ::ser::Serializer::new(writer, opts);
-            acceptor.accept(&mut serializer)
-        })
+    /// A trait for stopping serialization and deserialization when a certain limit has been reached.
+    pub trait SizeLimit: Clone {
+        /// Tells the SizeLimit that a certain number of bytes has been
+        /// read or written.  Returns Err if the limit has been exceeded.
+        fn add(&mut self, n: u64) -> Result<()>;
+        /// Returns the hard limit (if one exists)
+        fn limit(&self) -> Option<u64>;
+    }
+
+    pub trait BincodeByteOrder: Clone {
```

Comment:
`BincodeByteOrder` is only ever used as a type, not as an instance. This would make the `: Clone` constraint unneeded.